subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/DefaultConnection.java

```diff
@@ -42,7 +42,7 @@
 import org.gradle.tooling.internal.protocol.InternalCancellableConnection;
 import org.gradle.tooling.internal.protocol.InternalCancellationToken;
 import org.gradle.tooling.internal.protocol.InternalConnection;
-import org.gradle.tooling.internal.protocol.InternalFileSystemChangesAwareConnection;
+import org.gradle.tooling.internal.protocol.InternalNotifyDaemonsAboutChangedPathsConnection;
```

Comment:
This name is a bit weird. Not sure how to fix it. I like the ones that look like `InternalSomethingableConnection`.

Reply:
How about `InternalInvalidatableVirtualFileSystemConnection`?

Reply:
Sounds good!

Reply:
Renamed.